include/z64message.h

```diff
@@ -82,103 +211,103 @@ typedef struct {
     /* 0x11F1A */ s16 unk11F1A[3];
     /* 0x11F20 */ UNK_TYPE1 unk11F20[0x2];
-    /* 0x11F22 */ u8 msgMode; // TODO: MessageMode enum
+    /* 0x11F22 */ u8 msgMode;
     /* 0x11F23 */ UNK_TYPE1 unk11F23;
     /* 0x11F24 */ union {
         char schar[200];
         u16 wchar[100];
     } decodedBuffer;
     /* 0x11FEC */ u16 msgBufPos;
-    /* 0x11FEE */ s16 unk11FEE;
-    /* 0x11FF0 */ s16 unk11FF0;
-    /* 0x11FF2 */ u16 unk11FF2;
-    /* 0x11FF4 */ s16 unk11FF4;
-    /* 0x11FF6 */ s16 unk11FF6;
+    /* 0x11FEE */ u16 textDrawPos;
+    /* 0x11FF0 */ u16 decodedTextLen;
+    /* 0x11FF2 */ u16 textUnskippable;
+    /* 0x11FF4 */ s16 textPosX;
+    /* 0x11FF6 */ s16 textPosY;
     /* 0x11FF8 */ s16 unk11FF8;
     /* 0x11FFA */ s16 unk11FFA;
     /* 0x11FFC */ s16 unk11FFC;
     /* 0x11FFE */ s16 unk11FFE[0x3];
-    /* 0x12004 */ s16 unk12004;
-    /* 0x12006 */ s16 unk12006;
+    /* 0x12004 */ s16 textboxXTarget;
+    /* 0x12006 */ s16 textboxYTarget;
     /* 0x12008 */ s16 unk12008;
-    /* 0x1200A */ UNK_TYPE2 unk1200A;
+    /* 0x1200A */ s16 unk1200A;
     /* 0x1200C */ s16 unk1200C;
     /* 0x1200E */ s16 unk1200E;
     /* 0x12010 */ s16 unk12010;
     /* 0x12012 */ s16 unk12012;
     /* 0x12014 */ s16 unk12014;
     /* 0x12014 */ s16 unk12016;
-    /* 0x12018 */ s16 unk12018; // messageR
-    /* 0x1201A */ s16 unk1201A; // messageG
-    /* 0x1201C */ s16 unk1201C; // messageB
-    /* 0x1201E */ s16 unk1201E; // messageA
-    /* 0x12020 */ u8 unk12020;  // probably textboxEndType
+    /* 0x12018 */ s16 textColorR;
+    /* 0x1201A */ s16 textColorG;
+    /* 0x1201C */ s16 textColorB;
+    /* 0x1201E */ s16 textColorAlpha;
+    /* 0x12020 */ u8 textboxEndType;
     /* 0x12021 */ u8 choiceIndex;
-    /* 0x12022 */ u8 unk12022;
+    /* 0x12022 */ u8 choiceNum;
     /* 0x12023 */ u8 stateTimer;
-    /* 0x12024 */ s16 unk12024;
-    /* 0x12026 */ u16 unk12026;
+    /* 0x12024 */ u16 textDelayTimer;
+    /* 0x12026 */ u16 textDelay;
     /* 0x12028 */ u16 songPlayed;
     /* 0x1202A */ u16 ocarinaMode;
     /* 0x1202C */ u16 ocarinaAction;
     /* 0x1202E */ u16 lastPlayedSong;
     /* 0x12030 */ s16 unk_12030;
     /* 0x12032 */ UNK_TYPE1 unk_12032[0x2];
-    /* 0x12034 */ s16 unk12034;
-    /* 0x12036 */ s16 unk12036;
-    /* 0x12038 */ s16 unk12038;
-    /* 0x1203A */ s16 unk1203A;
-    /* 0x1203C */ s16 unk1203C;
+    /* 0x12034 */ s16 textboxColorRed;
+    /* 0x12036 */ s16 textboxColorGreen;
+    /* 0x12038 */ s16 textboxColorBlue;
+    /* 0x1203A */ s16 textboxColorAlphaTarget;
+    /* 0x1203C */ s16 textboxColorAlphaCurrent;
     /* 0x1203E */ s16 unk1203E;
-    /* 0x12040 */ struct Actor* unkActor;
+    /* 0x12040 */ struct Actor* talkActor;
     /* 0x12044 */ s16 unk12044;
-    /* 0x12046 */ s16 unk12046;
-    /* 0x12048 */ u8 unk12048; // EnKakasi
-    /* 0x12049 */ UNK_TYPE1 unk12049[0x1];
-    /* 0x1204A */ s16 unk1204A[0x5];
+    /* 0x12046 */ s16 blockSunsSong;
+    /* 0x12048 */ u8 lastOcarinaButtonIndex;
```

Comment:
```suggestion
    /* 0x12048 */ u8 ocarinaButtonIndex;
```
Actually, something I've been wanting to do on the main message branch is to drop the `last` here. I think `ocarinaButtonIndex` should be enough?